### ISSUE TITLE
FluentSender suppresses too many exceptions

### ISSUE THREAD

**Issue by miverson** [fluent/fluent-logger-python#55]
In _send_internal(), all exceptions are captured and suppressed, even if they are the result of a basic configuration error. This results in a total, silent failure of the library. 

In my case, I inadvertently set the port number to a string, rather than an int. Fluentd failed silently.

Ultimately, I modified the code to log the exception in _send_internal, and received:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/fluent/sender.py", line 97, in _send_internal
    self._reconnect()
  File "/usr/local/lib/python3.5/site-packages/fluent/sender.py", line 124, in _reconnect
    sock.connect((self.host, self.port))
TypeError: an integer is required (got type str)
```

I would suggest:
1. Add basic type checking to the constructor.
2. Revise the overly broad exception handling to only catch transient connection errors, allowing other exceptions to be handled appropriately by the calling code.

**Comment by repeatedly** [comment]
Sorry for the delay.
Your suggestion seems good.
I will work it later.

**Comment by repeatedly** [comment]
> 2.

From python document, using `socket.error` is better > https://docs.python.org/3.5/library/socket.html#socket.error

**Comment by arcivanov** [comment]
fixed in #57
